src/Router.php

```diff
@@ -121,16 +121,8 @@ public static function is_graphql_http_request() {
 		}
 
 		// If before 'init' check $_SERVER.
-		if ( isset( $_SERVER['SERVER_NAME'] ) && isset( $_SERVER['REQUEST_URI'] ) ) {
-			$haystack = wp_unslash( $_SERVER['SERVER_NAME'] )
-				. wp_unslash( $_SERVER['REQUEST_URI'] );
-			$needle   = site_url( self::$route );
-
-			// Strip protocol.
-			$haystack = preg_replace( '#^(http(s)?://)#', '', $haystack );
-			$needle   = preg_replace( '#^(http(s)?://)#', '', $needle );
-			$len      = strlen( $needle );
-			return ( substr( $haystack, 0, $len ) === $needle );
+		if ( isset( $_SERVER['SERVER_NAME'] ) && isset( $_SERVER['SERVER_PROTOCOL'] ) ) {
+			return ( strpos($_SERVER['SERVER_PROTOCOL'], 'HTTP') !== false );
```

Comment:
@sawanoboly @WybrenKoelmans 

Let me know if this solution works for ya'll:

```
	/**
	 * Returns true when the current request is a GraphQL request coming from the HTTP
	 *
	 * NOTE: This will only indicate whether the GraphQL Request is an HTTP request. Many features
	 * need to affect _all_ GraphQL requests, including internal requests using the `graphql()`
	 * function, so be careful how you use this to check your conditions.
	 *
	 * @return boolean
	 */
	public static function is_graphql_http_request() {
		
		/**
                  *Default value is false
                  *
                  * @param boolean $is_graphql_http_request Whether the request is a graphql HTTP request. Default false.
                  */
		$is_graphql_http_request = apply_filters( 'is_graphql_http_request', false );
		
		// Support wp-graphiql style request to /index.php?graphql.
		if ( true !== $is_graphql_http_request && isset( $_GET[ self::$route ] ) ) {
			$is_graphql_http_request = true;
		}

		// If before 'init' check $_SERVER.
		if ( true !== $is_graphql_http_request && isset( $_SERVER['HTTP_HOST'] ) && isset( $_SERVER['REQUEST_URI'] ) ) {
			$haystack = wp_unslash( $_SERVER['HTTP_HOST'] )
				. wp_unslash( $_SERVER['REQUEST_URI'] );
			$needle   = site_url( self::$route );

			// Strip protocol.
			$haystack = preg_replace( '#^(http(s)?://)#', '', $haystack );
			$needle   = preg_replace( '#^(http(s)?://)#', '', $needle );
			$len      = strlen( $needle );
			$is_graphql_http_request = ( substr( $haystack, 0, $len ) === $needle );
		}

		return $is_graphql_http_request;
	}

```